client/src/components/game/gui/panels/map/MapPanel.jsx

```diff
@@ -1,10 +1,74 @@
-import React from "react";
+import React, { useState } from "react";
 import PropTypes from "prop-types";
+import {
+    MapContainer, TileLayer, Marker, useMapEvents, Tooltip,
+} from "react-leaflet";
+import Leaflet, { CRS } from "leaflet";
 import PanelTemplate from "../panel_template/PanelTemplate";
 import mapIcon from "../../../../../assets/images/gui/hud/map-icon.png";
+import playerIcon from "../../../../../assets/images/gui/panels/map/player-marker.png";
+import destinationIcon from "../../../../../assets/images/gui/panels/map/destination-marker.png";
+import "leaflet/dist/leaflet.css";
 import "./MapPanel.scss";
+import { ApplicationState, PlayerState } from "../../../../../shared/state/States";
+
+const markerScale = 4;
+
+const playerMarker = Leaflet.icon({
+    iconUrl: playerIcon,
+    iconSize: [12 * markerScale, 14 * markerScale],
+    iconAnchor: [24, 48],
+});
+
+const locationMarker = Leaflet.icon({
+    iconUrl: destinationIcon,
+    iconSize: [12 * markerScale, 14 * markerScale],
+    iconAnchor: [24, 48],
+});
+
+function LocationMarker() {
+    const [position, setPosition] = useState(null);
+
+    useMapEvents({
+        click(e) {
+            let { lat, lng } = e.latlng;
+
+            lat = parseInt(lat, 10);
+            lng = parseInt(lng, 10);
+
+            let newPosition = { lat, lng };
+
+            // remove marker if previous position is same as new position
+            if (position !== null) {
+                if (position.lat === newPosition.lat
+                && position.lng === newPosition.lng) {
+                    newPosition = null;
+                }
+            }
+
+            setPosition(newPosition);
+        },
+    });
+
+    return position === null ? null : (
+        <Marker position={position} icon={locationMarker}>
+            <Tooltip className="press-start-font tooltip" direction="top" offset={[0, -44]} opacity={1} permanent>
+                { `${position.lat}, ${position.lng}` }
+            </Tooltip>
+        </Marker>
+    );
+}
 
 function MapPanel({ onCloseCallback }) {
+    const leafletConfig = {
+        center: [
+            // The center is a decimal (with the extra half tile), so round down for display.
+            Math.floor(-PlayerState.row + 32),
+            Math.floor(PlayerState.col - 248 + 0.5),
```

Comment:
Doing the Math.floor here rounds the number down, which makes the + 0.5 have no effect.